### PR TITLE
Add serialisation test for temporal models

### PR DESCRIPTION
**Pull request by QRemy**
Follow up of PR #3574
Add serialisation tests for temporal models.

@bkhelifi, note that Model.from_dict() works only for  specific classes.
So if one call model_dict = PowerLawTemporalModel.to_dict()
 then PowerLawTemporalModel.from_dict(model_dict) works but TemporalModel.from_dict(model_dict) create a TemporalModel and not a PowerLawTemporalModel. To prevent this I could raise an error in from_dict if `data['type'] not in cls.tag` (in a separate PR).

### CODE REVIEW

**Comment by bkhelifi** [comment]
If I understand well, one needs to know in advance the type of the model in order to read it correctly? If true, it is not very flexible, is not it?

File: gammapy/modeling/models/tests/test_temporal.py [hunk 1]
```diff
@@ -288,3 +295,21 @@ def test_plot_constant_model():
     constant_model = ConstantTemporalModel(const=1)
     with mpl_plot_check():
         constant_model.plot(time_range)
+
+
+def test_io():
```
Comment (adonath) [inline gammapy/modeling/models/tests/test_temporal.py:300]: Minor: maybe move to `gammapy/modeling/models/test_io.py` instead? This where the other I/O tests are...

**Review by adonath — APPROVED** [review]
Thanks @QRemy, I have left one minor comment on the test location. Otherwise looks good to me...

File: gammapy/modeling/models/tests/test_temporal.py (hunk 1 above)
Comment (QRemy) [reply gammapy/modeling/models/tests/test_temporal.py]: done

**Review by adonath — APPROVED** [review]
Thanks @QRemy!

**Pull request merged**